core/parameters/src/config_store.rs

```diff
@@ -105,6 +107,24 @@ impl RuntimeConfigStore {
                 store.insert(*protocol_version, Arc::new(runtime_config));
             }
         }
+        if is_benchmarknet {
```

Comment:
I wonder if we can avoid adding another argument `is_benchmarknet` to the constructor and do the required patching outside, e.g. in the `for_chain_id` function...

Reply:
As I see, `RuntimeConfig` stores config variables into multiple logic-separated set, such that `fees`, `wasm_config`, etc, and some of this subconfigs also divided into smaller ones. OTOH, `.yaml` diff files store all variables together, while logic of separating them is hidden into [this `try_from`](https://github.com/near/nearcore/blob/dae8e8476b6e101351b394d9ca4f57a79512b234/core/parameters/src/parameter_table.rs#L292) function.

Thus, while operating with unfinished `RuntimeConfigStore`, we can reuse `ParameterTable::apply_diff` function which easily does all what we want.


If we want change parameters after `RuntimeConfigStore` is already created, I see a couple of ways:

- hardcode to which subconfig in configs tree each parameter should go into `for_chain_id` function will go, which sounds to look monstrous (and kinda code duplicating with `try_from`)
- add recoursive-style method `to_parameter_table` for each subconfig, which adds some not very necessary methods to a number of structs.

With everything above, since we already have code with this new argument, will adding comment like "if you don't want what this argument should be, set it to false" good enough?

Reply:
@akashin do you think this is crucial and merge-blocking or we can keep it as is?

Reply:
> hardcode to which subconfig in configs tree each parameter should go into for_chain_id function will go, which sounds to look monstrous (and kinda code duplicating with try_from)

I think the first option option should be simple to implement. I'll see if I can get it working.

Considering the number of changes in other files that use this `new` method, I would strongly prefer not to touch it.

Reply:
Here is the example of how to do this: https://github.com/near/nearcore/compare/akashin/benchmark_override?expand=1

True, a bit ugly due to need to work with `Arc`s, but I think it's a simpler alternative.